api/handler/attributes.go

```diff
@@ -98,25 +98,31 @@ func (h *handler) GetObjectAttributesHandler(w http.ResponseWriter, r *http.Requ
 		return
 	}
 
+	bktSettings, err := h.obj.GetBucketSettings(r.Context(), bktInfo)
+	if err != nil {
+		h.logAndSendError(w, "could not get bucket settings", reqInfo, err)
+		return
+	}
+
 	response, err := encodeToObjectAttributesResponse(info, params)
 	if err != nil {
 		h.logAndSendError(w, "couldn't encode object info to response", reqInfo, err)
 		return
 	}
 
-	writeAttributesHeaders(w.Header(), info, params)
+	writeAttributesHeaders(w.Header(), extendedInfo, bktSettings.Unversioned())
 	if err = api.EncodeToResponse(w, response); err != nil {
 		h.logAndSendError(w, "something went wrong", reqInfo, err)
 	}
 }
 
-func writeAttributesHeaders(h http.Header, info *data.ObjectInfo, params *GetObjectAttributesArgs) {
-	h.Set(api.LastModified, info.Created.UTC().Format(http.TimeFormat))
-	if len(params.VersionID) != 0 {
+func writeAttributesHeaders(h http.Header, info *data.ExtendedObjectInfo, isBucketUnversioned bool) {
+	h.Set(api.LastModified, info.ObjectInfo.Created.UTC().Format(http.TimeFormat))
+	if !isBucketUnversioned {
 		h.Set(api.AmzVersionID, info.Version())
 	}
 
-	if info.IsDeleteMarker {
+	if info.NodeVersion.DeleteMarker != nil {
```

Comment:
Maybe we should add a new method `IsDeleteMarker` to `NodeVersion` struct and compare the `DeleteMarker` field with `nil` there